wallet/src/hooks/User/index.jsx

```diff
@@ -100,14 +116,40 @@ export const UserProvider = ({ children }) => {
             await acc; // Acc is a promise so we await it before processing the next one;
             return blockProposedEventHandler(curr, [zkpPrivateKey], [nullifierKey]); // TODO Should be array
           }, Promise.resolve());
+
+        // We want to verify that the received block is from the current contract deployment.
+        // for this, we store the lastBlock received in lastBlock, and compare the hash with the previousBlockHash
+        // from the received block. If they don't match, then there has been a redeployment.
+        // Additionally, before a transfer/withdraw, there is a check the the last block stored hash, lasbBlockDb
+        // matches the blockChain.
         if (Number(parsed.maxBlock) !== 1) {
-          socket.send(
-            JSON.stringify({
-              type: 'sync',
-              lastBlock:
-                parsed.historicalData[parsed.historicalData.length - 1].block.blockNumberL2,
-            }),
-          );
+          const lastBlockDb = await getLastBlock();
```

Comment:
Hey @druiz0992, the method `getLastBlock` returns `null` if no block is found and the variable is not tested against this situation it might cause an issue.